### ISSUE TITLE
If validation period elapsed, use pending aggregate root in proveAndProcess

### ISSUE THREAD

**Issue by jakekidd** [connext/nxtp#1963]
**Describe the bug**
When we call `proveAndProcess`, we should be able to accept using the `pendingAggregateRoot` for proofs IFF the root has already been validated (i.e. the validation period in blocks has elapsed).

There isn't a clean cut resolution here, but a good compromise is probably in order. Some constraints:
- We shouldn't use the pendingAggregateRoot just because the validation period has elapsed. What if a relayer sends a proof tx just before the validation period elapsed? The proof - which will probs be for the currentAggregateRoot - will fail to get the pendingAggregateRoot.
- We shouldn't just switch up pendingAggregateRoot and currentAggregateRoot blindly, since that will potentially rug/grief other relayer txs in the mempool (and therefore users)
- It would suck to have a ton of separate calls just check to see that the current pendingAggregateRoot is verified, and then not do anything about it... at that point we might as well go ahead and label it as proven.

TL;DR: Some racey problems with state updating and relayer txs.

One option:
- Keep a mapping of `provenAggregateRoots`
- Caller of `proveAndProcess` supplies `bytes32 _targetAggregateRoot`, indicating which they plan on using.
- We look up the target aggregate root in `provenAggregateRoots`
- If it's not there, then we can check whether it's the `pendingAggregateRoot`; if so, check to see if the validation period has elapsed for that pending root.
    - If we use the pending and the validation period has elapsed, go ahead and add the `pendingAggregateRoot` to `provenAggregateRoots`

Additionally: would be nice to have a separate public method that literally just does that last bit: check to see if validation period has elapsed and if so, move pending root -> proven roots.

(Finally: would be great to have an optimizer that allows the caller of the proveAndProcess call to specify that the root is *definitely the pending one* so we can skip checking the proven roots and assume it's not there... but that could be racey, so idk)
    
One lingering concern: is it okay to have a mapping of proven roots for the aggregate roots on the spoke chains?

See discussion:
https://discord.com/channels/992549151134982234/1022516954986184794/1024079262489247917
